CMakeLists.txt

```diff
@@ -50,6 +63,11 @@ if(MINGW)
   set(EXTRA_LIBS ${EXTRA_LIBS} ws2_32)
   set(TEST_LIBS ${TEST_LIBS} ws2_32)
 endif()
+# BSD doesn't link pthread by default, so make it explicit
+if(NOT WIN32)
+  set(EXTRA_LIBS ${EXTRA_LIBS} pthread)
+  set(TEST_LIBS ${TEST_LIBS} pthread)
+endif()
```

Comment:
I don't know why you think threading is part of libc on FreeBSD. Historically you could choose your threading implementation between one of three options, 1:1 threading, M:N threading or "system default" which was a symlinked libthread.so to one of the previous two. Therefore to link to some threading library don't care which you did an explicit -lpthread.

The -pthread facility is buggy on FreeBSD, or at least it was. Also, your choice of threading libraries is now exactly one, the 1:1 library, and libpthread always maps to that. A quick search of google confirms that on FreeBSD, the best practice method of linking against pthreads is -lpthread.

I can make the if(NOT WIN32) to something like if(BSD), but I'll need to add extra cmake scripting to detect BSD. Let me know if that works for you.